processor/deltatorateprocessor/config.go

```diff
@@ -26,6 +27,41 @@ type Config struct {
 
 	// List of delta sum metrics to convert to rates
 	Metrics []string `mapstructure:"metrics"`
+
+	// Time unit to calculate the rate over. This could be one of the [Nanosecond, Millisecond, Second, Minute]. Default is Second.
+	TimeUnit StringTimeUnit `mapstructure:"time_unit"`
+}
+
+type StringTimeUnit string
+
+const (
+	nanosecond  StringTimeUnit = "Nanosecond"
+	millisecond StringTimeUnit = "Millisecond"
+	second      StringTimeUnit = "Second"
+	minute      StringTimeUnit = "Minute"
+)
+
+var timeUnits = map[StringTimeUnit]struct{}{
+	nanosecond:  {},
+	millisecond: {},
+	second:      {},
+	minute:      {},
+}
+
+var validTimeUnits = func() []string {
+	ret := make([]string, len(timeUnits))
+	i := 0
+	for k := range timeUnits {
+		ret[i] = string(k)
+		i++
+	}
+	sort.Strings(ret)
+	return ret
+}
```

Comment:
Any specific reason that you define a `timeUnits` map and then convert it to `array`. Can you directly use define the array with the valid `StringTimeUnit`?

Reply:
Hi Min, thanks for the review. Map is faster to lookup. Thats why the default is a map. This array is only used for error cases to print the message properly. Actually, this was suggested by Anthony for my previous PR. I want to stay with this if you don't think its a major concern.